File: LeetCode/566/main.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        row, col = len(mat[0]), len(mat)
        if(row * col != r * c) :
            return mat
        
        ans = [[0] * c for _ in range(r)]
        
        for i in range(row * col):
            ans[i // c][i % c] = mat[i // row][i % row]
            
        return ans

File: LeetCode/566/main.cpp
class Solution {
public:
  vector<vector<int>> matrixReshape(vector<vector<int>> &arr, int r, int c) {

    int n = arr.size();
    int m = arr[0].size();

    if (r * c != m * n) {
      return arr;
    }

    vector<vector<int>> mat(r, vector<int>(c, 0));
    int row = 0, col = 0;

    for (int i = 0; i < n; i++) {
      for (int j = 0; j < m; j++) {
        if (col > c - 1) {
          col = 0;
          row++;
        }
        mat[row][col] = arr[i][j];
        col++;
      }
    }

    /*
        OR

        int count = 0

        for(int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                mat[count / c][count % c] = arr[i][j];
                count++;
            }
        }
    */

    return mat;
  }
};